src/controllers/controller.doughnut.js

```diff
@@ -19,25 +19,27 @@ defaults._set('doughnut', {
 		mode: 'single'
 	},
 	legendCallback: function(chart) {
-		var text = [];
-		text.push('<ul class="' + chart.id + '-legend">');
-
+		var list = document.createElement('ul');
 		var data = chart.data;
 		var datasets = data.datasets;
-		var labels = data.labels;
 
 		if (datasets.length) {
-			for (var i = 0; i < datasets[0].data.length; ++i) {
-				text.push('<li><span style="background-color:' + datasets[0].backgroundColor[i] + '"></span>');
+			var labels = data.labels;
```

Comment:
you should declare these above the for loop. the chart.js coding style is to declare all variables next to each other at the beginning of a method

Reply:
got it. my motivation here was that the vars aren't needed if the condition is false. wasn't sure if that should be taken into account before declaring them. I can add a commit to move them right now though.